### ISSUE TITLE
Fix macOS OpenSSL instructions

### ISSUE THREAD

**Issue by joaomoreno** [Microsoft/vsts-agent#969]
Due to #470, the official docs recommend to replace OpenSSL in a system-wide fashion:

https://github.com/Microsoft/vsts-agent/blob/master/docs/start/envosx.md

Almost a year ago @joshka reported that this is dangerous and it will _leave your system in an inconsistent state_. @davibe also tried to address this by using the `DYLD_LIBRARY_PATH` variable.

It turns out that this recommendation is a really bad idea. It will cause strange issues, as it did in our build agents. This took me way too long to figure out and could've totally been avoided.

@bryanmacfarlane's suggestion seems to work much better. I've ran this on our agents:

```
install_name_tool -add_rpath /usr/local/opt/openssl/lib ./bin/System.Security.Cryptography.Native.OpenSsl.dylib
```

This keeps the use of Homebrew's OpenSSL strictly to .NET, leaving everything else in your system untouched.

Please consider changing your documentation to use this instead of the current solution.

**Comment by TingluoHuang** [comment]
@joaomoreno with netcore 2.0, the agent will use Apple crypto libraries instead of OpenSSL. I will close this issue once we adopt netcore 2.0.

**Comment by TingluoHuang** [comment]
We released the agent using netcore 2.0, Openssl is no longer required for netcore 2.0.
https://github.com/Microsoft/vsts-agent/releases/tag/v2.125.0
